exchanges/cryptodotcom/cryptodotcom_websocket.go

```diff
@@ -0,0 +1,613 @@
+			if !cr.IsFillsFeedEnabled() {
+				return nil
+			}
+			return cr.processUserTrade(resp.Result)
+		case strings.HasPrefix(resp.Result.Channel, userBalanceCnl):
+			return cr.processUserBalance(resp.Result)
+		case strings.HasPrefix(resp.Result.Channel, instrumentOrderbookCnl):
+			return cr.processOrderbook(resp.Result)
+		case strings.HasPrefix(resp.Result.Channel, tickerCnl):
+			return cr.processTicker(resp.Result)
+		case strings.HasPrefix(resp.Result.Channel, tradeCnl):
+			return cr.processTrades(resp.Result)
+		case strings.HasPrefix(resp.Result.Channel, candlestickCnl):
+			return cr.processCandlestick(resp.Result)
+		default:
+			if resp.Code == 0 {
+				return nil
+			}
+		}
+	}
+	if !cr.Websocket.Match.IncomingWithData(resp.ID, respRaw) {
+		return fmt.Errorf("could not match incoming message with signature: %d, and data: %s", resp.ID, string(respRaw))
+	}
+	return nil
+}
+
+func (cr *Cryptodotcom) processCandlestick(resp *WsResult) error {
+	var data []WsCandlestickItem
+	err := json.Unmarshal(resp.Data, &data)
+	if err != nil {
+		return err
+	}
+	cp, err := currency.NewPairFromString(resp.InstrumentName)
+	if err != nil {
+		return err
+	}
+	interval, err := stringToInterval(resp.Interval)
+	if err != nil {
+		return err
+	}
+	candles := make([]stream.KlineData, len(data))
+	for x := range data {
+		candles[x] = stream.KlineData{
+			Pair:      cp,
+			Exchange:  cr.Name,
+			Timestamp: data[x].UpdateTime.Time(),
+			Interval:  interval.Word(),
+			AssetType: asset.Spot,
+			OpenPrice: data[x].Open,
+			HighPrice: data[x].High,
+			LowPrice:  data[x].Low,
+			Volume:    data[x].Volume,
+			StartTime: data[x].EndTime.Time(), // This field represents Start Timestamp for websocket push data. and End Timestamp for REST
+		}
+	}
+	cr.Websocket.DataHandler <- candles
+	return nil
+}
+
+func (cr *Cryptodotcom) processTrades(resp *WsResult) error {
+	var data []TradeItem
+	err := json.Unmarshal(resp.Data, &data)
+	if err != nil {
+		return err
+	}
+	cp, err := currency.NewPairFromString(resp.InstrumentName)
+	if err != nil {
+		return err
+	}
+	trades := make([]trade.Data, len(data))
+	for i := range data {
+		var oSide order.Side
+		oSide, err = order.StringToOrderSide(data[i].Side)
+		if err != nil {
+			return err
+		}
+		trades[i] = trade.Data{
+			Amount:       data[i].TradeQuantity,
+			Price:        data[i].TradePrice,
+			AssetType:    asset.Spot,
+			CurrencyPair: cp,
+			Exchange:     cr.Name,
+			Side:         oSide,
+			Timestamp:    data[i].TradeTimestamp.Time(),
+			TID:          data[i].TradeID,
+		}
+	}
+	return trade.AddTradesToBuffer(cr.Name, trades...)
+}
+
+func (cr *Cryptodotcom) processTicker(resp *WsResult) error {
+	var data []TickerItem
+	err := json.Unmarshal(resp.Data, &data)
+	if err != nil {
+		return err
+	}
+	cp, err := currency.NewPairFromString(resp.InstrumentName)
+	if err != nil {
+		return err
+	}
+	tickersDatas := make([]ticker.Price, len(data))
```

Comment:
This doesn't do anything because `websocketroutine_manager.go` has no way of handling `[]ticker.Price` so the data goes nowhere, forcing a REST switchover.

You'll need to update `websocketroutine_manager.go` to support it in the function `websocketDataHandler`